### PR TITLE
Fixed disabled switch visual edit when json is empty

### LINKED ISSUE #2957: Unable to go back to the visual editor after clicking on "Switch to JSON editor" when creating a new role mapping
| Wazuh | Elastic | Rev |
| ----- | ------- | --- |
| 4.1.0   | 7.10.2    | 4101-1 |

**Description**
When creating a new role mapping if the user clicks on "Switch to JSON editor" then is unable to go back to the visual editor as a message appears "Current rule can't be edited using visual editor"

output

### PR DESCRIPTION
**Pull request by Machi3mfl**
Hi team, this resolves

 - Fixed disabled switch visual editor button in Rules mapping editor

Closes #2957

### CODE REVIEW

**Review by gabiwassan — APPROVED** [review]
LGTM!

**Comment by Desvelao** [comment]
I tested the PR:

If the editor has:
- `{}`: the user can switch to the visual editor
- empty: the user can not switch to the visual editor. I am not sure if this behavior is we want.
image

**Comment by Desvelao** [comment]
The change in `CHANGELOG.md` is missing. You should add it.

File: public/components/security/roles-mapping/components/rule-editor.tsx [hunk 1]
```diff
@@ -111,17 +111,30 @@ export const RuleEditor = ({ save, initialRule, isLoading, isReserved, internalU
     rulesTmp.splice(id, 1);
     setRules(rulesTmp);
   };
+  
+  const getRulesFromJson = (jsonRule) => {
+    if (jsonRule !== '{}') {
```
Comment (pablomarga) [inline public/components/security/roles-mapping/components/rule-editor.tsx:116]: I tested this PR and the behaviour that @Desvelao has commented would be solved changing `if (jsonRule !== '{}')`  to ` if (jsonRule !== '') `

**Comment by Desvelao** [comment]
I tested the PR and now it lets switch to the visual editor if the JSON content editor is empty or `{}`. Good job! :+1:
image
image

**Pull request merged**